tools/generator/src/Generator/SetTargetConfigurations.swift

```diff
@@ -287,13 +287,21 @@ Target with id "\(id)" not found in `consolidatedTarget.uniqueFiles`
             buildSettings["MACH_O_TYPE"] = "staticlib"
         }
 
-        let compileTargetName: String
-        if let compileTarget = target.compileTarget {
+        if let target_ids = target.compileTargets?.map(\.id.rawValue) {
             buildSettings.set(
-                "BAZEL_COMPILE_TARGET_ID",
-                to: compileTarget.id.rawValue
+                "BAZEL_COMPILE_TARGET_IDS",
+                to: target_ids
             )
-            compileTargetName = compileTarget.name
+        }
+
+        let compileTargetName: String
+        if let compileTargets = target.compileTargets, compileTargets.count > 0 {
+            // Intentionally take the first `compileTargets` entry since this
+            // will be Swift in the case where a Swift compile target is present
+            // amongst a non-Swift compile target. Swift indexstore imports are
+            // the only supported ones right now so this opts to choose that when
+            // possible.
+            compileTargetName = compileTargets[0].name
```

Comment:
@brentleyjones, when you have a moment, please make sure you feel this comment is valid for what it represents. This is "used" right now since there's no multi-compile targets provided but it will be in the future.